cluster-autoscaler/utils/kubernetes/listers.go

```diff
@@ -144,18 +144,63 @@ type PodLister interface {
 	List() ([]*apiv1.Pod, error)
 }
 
+func isScheduled(pod *apiv1.Pod) bool {
+	if pod == nil {
```

Comment:
Why do we need to nil check? Shouldn't `pod` always be non-nil in these functions?

Reply:
Yes, but it can be used in other functions as well, so it'd be safer to check it inside of the function.

Reply:
Yes, but it is private so these potential future functions would be in the same module, which limits the risk.

Reply:
Updated & added a note comment for callers, still not sure about that decision though :')